components/resourceDetails/resourceSharing/addAgentButton/addAgentRow/index.jsx

```diff
@@ -89,14 +90,21 @@ export default function AddAgentRow({
   const classes = useStyles();
   const [agentWebId, setAgentWebId] = useState("");
   const [existingPermission, setExistingPermission] = useState();
-
-  const agentName = (thing && getStringNoLocale(thing, foaf.name)) || null;
-  const agentAvatar = (thing && getUrl(thing, vcard.hasPhoto)) || null;
-  const displayedWebId =
-    (thing &&
-      contactsArrayLength > 0 &&
-      getUrl(thing, VCARD_WEBID_PREDICATE)) ||
-    null;
+  const [agentName, setAgentName] = useState(null);
```

Comment:
It is sometimes a contact but it can also be the temporary thing we are adding to the dataset by adding new webId, which is why I didn't go with contact.

Reply:
But the new thing should also represent a contact (albeit a contact in the making)? I guess we could make it explicit, e.g. `contactOrContactToBeCreated`, but that's a bit too verbose. Ok, as long `thing` is easy to understand based on its context, it should be ok.

Reply:
It can be a contact, or a contact to be created, or not if it cannot find a profile, so I named it `temporaryRowThing` to make it a bit more readable.